### PR TITLE
Update AJCC validation script to be an IF-andonly-IF

### PR DESCRIPTION
**Pull request by wajiha-oicr**
Apologize in advance for huge diff, prettier fixed a bunch of whitespace

Before:
- the t, n, m fields were required if the main field was an AJCC field
 
After:
- the t,n,m fields are required IF and ONLY if the main field was and AJCC field
    - compares the length of the emptyFields array; if it isn't the same length as the array of the fields we're checking for, we know they've been wrongfully provided. compute the difference of the two arrays and use in the error message.

Other: 
- Added and updated unit tests
- Updated a dependency ( to fix vulnerability)

### CODE REVIEW

File: references/validationFunctions/common/ajccValidation.js [hunk 1]
```diff
@@ -1,37 +1,56 @@
...
+          (!convertedRow[field] || checkforEmpty(convertedRow[field])),
+      );
+
+      /* The fields should be provided IF and ONLY IF the AJCC regex passes */
+      if (/^(AJCC)\b/i.test($field) && emptyFields.length) {
+        result = {
+          valid: false,
+          message: `The following fields are required when ${$name} is set to an AJCC option: ${emptyFields}`,
+        };
+      } else if (!/^(AJCC)\b/i.test($field) && emptyFields.length != requiredFields.length) {
```
Comment (wajiha-oicr) [inline references/validationFunctions/common/ajccValidation.js]: this is the additional condition added to this script

File: tests/common/ajccValidation.test.js [hunk 2]
```diff
@@ -2,227 +2,265 @@ const validation = require('./../../references/validationFunctions/common/ajccVa
...
+      true,
+      {
+        row: loadObjects(specimen, {
+          pathological_tumour_staging_system: 'Aajccc fake edition',
+        }),
+        name: 'pathological_tumour_staging_system',
+      },
+    ],
+    [
+      'Non AJCC Value with all categories set.',
```
Comment (wajiha-oicr) [inline tests/common/ajccValidation.test.js]: one of the tests to ensure the new condition works

File: tests/common/ajccValidation.test.js [hunk 3]
```diff
@@ -2,227 +2,265 @@ const validation = require('./../../references/validationFunctions/common/ajccVa
...
+      true,
+      {
+        row: loadObjects(followUp, {
+          posttherapy_tumour_staging_system: 'Lugano',
+        }),
+        name: 'posttherapy_tumour_staging_system',
+      },
+    ],
+    [
+      'Post-Therapy: Non AJCC Value with <t,n,m>_category values set',
```
Comment (wajiha-oicr) [inline tests/common/ajccValidation.test.js]: another test

File: tests/common/ajccValidation.test.js [hunk 4]
```diff
@@ -2,227 +2,265 @@ const validation = require('./../../references/validationFunctions/common/ajccVa
...
+      'Recurrence: Non AJCC Value with <t,n,m>_category values not set.',
+      true,
+      {
+        row: loadObjects(followUp, {
+          recurrence_tumour_staging_system: 'FIGO',
+        }),
+        name: 'recurrence_tumour_staging_system',
+      },
+    ],
+    [
```
Comment (wajiha-oicr) [inline tests/common/ajccValidation.test.js]: another test that has 2 out of the 3 fields wrongfully provided

File: tests/common/ajccValidation.test.js [hunk 5]
```diff
@@ -2,227 +2,265 @@ const validation = require('./../../references/validationFunctions/common/ajccVa
...
+      true,
+      {
+        row: loadObjects(primaryDiag, {
+          clinical_tumour_staging_system: 'Ann Arbor',
+        }),
+        name: 'clinical_tumour_staging_system',
+      },
+    ],
+    [
+      'Non AJCC Value with <t>_category values  set.',
```
Comment (wajiha-oicr) [inline tests/common/ajccValidation.test.js]: another test with 1/3 fields wrongfully provided

File: references/validationFunctions/common/ajccValidation.js (hunk 1 above)
Comment (joneubank) [reply references/validationFunctions/common/ajccValidation.js]: This is so helpful!

Comment (wajiha-oicr) [reply references/validationFunctions/common/ajccValidation.js]: All of these files existing before I started using the prettier extension so they all create huge diffs and it's hard to tell what changed :(

It's honestly the least I could do

File: references/validationFunctions/common/ajccValidation.js [hunk 6]
```diff
@@ -1,37 +1,56 @@
...
+        result = {
+          valid: false,
+          message: `The following fields cannot be provided when ${$name} is not set to an AJCC option: ${requiredFields.filter(
```
Comment (joneubank) [inline references/validationFunctions/common/ajccValidation.js]: Do we want to format this message in the same way as other messages, with a list starting with dashes? right now it looks like this shows an array in a square bracket...

Just want to be consistent.

Comment (wajiha-oicr) [reply references/validationFunctions/common/ajccValidation.js]: Yeah that's a good point! I'll do the list with dashes for this message.

**Comment by wajiha-oicr** [comment]
> Code looks good, just make sure the output message is consistent with others. If it is then this is good to go.

thanks for the review Jon. I applied the same error formatting technique to the script in the new commit https://github.com/icgc-argo/argo-dictionary/pull/81/commits/0b5143e3b7c3585a4a746f7408a01347d4c87872

**Pull request merged**